alexandria/core/serializers.py

```diff
@@ -219,6 +221,12 @@ def validate(self, *args, **kwargs):
                 f'"original" must not be set for variant "{variant}".'
             )
 
+        mime_type = validated_data["content"].content_type
+        if mime_type == "application/octet-stream":
+            guess, _ = guess_type(validated_data["name"])
+            if guess is not None:
+                mime_type = guess
+        validated_data["mime_type"] = mime_type
         validated_data["mime_type"] = validated_data["content"].content_type
```

Comment:
Is this intended to be left? doesnt this overwrite the changes just made

Reply:
Oops that is a leftover indeed